test/integration/noderesourcetopology_test.go

```diff
@@ -0,0 +1,636 @@
+	testutils "k8s.io/kubernetes/test/integration/util"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	scheconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+	"sigs.k8s.io/scheduler-plugins/pkg/noderesourcetopology"
+	"sigs.k8s.io/scheduler-plugins/test/util"
+)
+
+func TestTopologyMatchPlugin(t *testing.T) {
+	todo := context.TODO()
+	ctx, cancelFunc := context.WithCancel(todo)
+	testCtx := &testutils.TestContext{
+		Ctx:      ctx,
+		CancelFn: cancelFunc,
+		CloseFn:  func() {},
+	}
+	registry := fwkruntime.Registry{noderesourcetopology.Name: noderesourcetopology.New}
+	t.Log("create apiserver")
+	_, config := util.StartApi(t, todo.Done())
+
+	config.ContentType = "application/json"
+
+	apiExtensionClient, err := apiextensionsclient.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeConfigPath := util.BuildKubeConfigFile(config)
+	if len(kubeConfigPath) == 0 {
+		t.Fatal("Build KubeConfigFile failed")
+	}
+	defer os.RemoveAll(kubeConfigPath)
+
+	t.Log("create crd")
+	if _, err := apiExtensionClient.ApiextensionsV1().CustomResourceDefinitions().Create(ctx, makeNodeResourceTopologyCRD(), metav1.CreateOptions{}); err != nil {
+		t.Fatal(err)
+	}
+
+	cs := kubernetes.NewForConfigOrDie(config)
+
+	topologyClient, err := topologyclientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = wait.Poll(100*time.Millisecond, 3*time.Second, func() (done bool, err error) {
+		groupList, _, err := cs.ServerGroupsAndResources()
+		if err != nil {
+			return false, nil
+		}
+		for _, group := range groupList {
+			if group.Name == "topology.node.k8s.io" {
+				return true, nil
+			}
+		}
+		t.Log("waiting for crd api ready")
+		return false, nil
+	}); err != nil {
+		t.Fatalf("Waiting for crd read time out: %v", err)
+	}
+
+	ns, err := cs.CoreV1().Namespaces().Create(ctx, &v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("integration-test-%v", string(uuid.NewUUID()))}}, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to integration test ns: %v", err)
+	}
+
+	autoCreate := false
+	t.Logf("namespaces %+v", ns.Name)
+	_, err = cs.CoreV1().ServiceAccounts(ns.Name).Create(ctx, &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{Name: "default", Namespace: ns.Name}, AutomountServiceAccountToken: &autoCreate}, metav1.CreateOptions{})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("Failed to create ns default: %v", err)
+	}
+
+	testCtx.NS = ns
+	testCtx.ClientSet = cs
+
+	args := &scheconfig.NodeResourceTopologyMatchArgs{
+		KubeConfigPath: kubeConfigPath,
+		Namespaces:     []string{ns.Name},
+	}
+
+	profile := schedapi.KubeSchedulerProfile{
+		SchedulerName: v1.DefaultSchedulerName,
+		Plugins: &schedapi.Plugins{
+			PreBind: &schedapi.PluginSet{
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
```

Comment:
This disabling seems irrelevant (also applies to Filter, PreFilter, Reserve).

(correct me if I'm wrong)

Reply:
w/o disabling PreBind VolumeBinding's PreBing plugin is getting called, probably it's due to VolumeBinding in algorithmprovider's registry and it goes to defaultPlugins.
I rechecked, Reserve & PreFilter now plugins don't fail, maybe add volume to our test, in this case VolumeBinding probably should pass.

Reply:
I'd expect this plugin to co-exist with default in-tree plugins, otherwise, users would also run into problems as this integration test. So could you update the test to remove all irrelevant disabling?

Reply:
I don't think it's related to PreBind of VolumeBinding. The conflict seems to hide in some `Filter` plugin.

Reply:
ok, it's because the memory was built with plain "100", you should use "100Gi".